tests/tests/test-precompile/test-precompile-xtokens.ts

```diff
@@ -158,7 +159,7 @@ describeDevMoonbeamAllEthTxTypes("Precompiles - xtokens", (context) => {
   });
 });
 
-describeDevMoonbeamAllEthTxTypes("Precompiles - xtokens", (context) => {
+describeDevMoonbeam("Precompiles - xtokens", (context) => {
```

Comment:
Is this intentional because it takes EIP-1559 by default?

Reply:
Yes, so the problem is the way `createTransaction` handles `maxPriorityFeePerGas`, which I think could be improved (I'll open an internal issue to track that).

Basically, in the case of (1) EIP-1559 txn with (2) `gasPrice` provided but no `maxFeePerGas`, it will specify `maxPriorityFeePerGas: 0`, which leads to a discrepancy in the exact fees charged between txn types:

```typescript
export const createTransaction = async (
  context: DevTestContext,
  options: TransactionOptions
): Promise<string> => {
  const isLegacy = context.ethTransactionType === "Legacy";
  const isEip2930 = context.ethTransactionType === "EIP2930";
  const isEip1559 = context.ethTransactionType === "EIP1559";

  const gasPrice = options.gasPrice !== undefined ? options.gasPrice : DEFAULT_TXN_MAX_BASE_FEE;
  const maxPriorityFeePerGas =
    options.maxPriorityFeePerGas !== undefined ? options.maxPriorityFeePerGas : 0;
```